### ISSUE TITLE
Impossible to display a image using Gtk on macOSX

### ISSUE THREAD

**Issue by mgautierfr** [pyinstaller/pyinstaller#2669]
With a simple Gtk project creating a window and displaying an image.
The packaged application fails 

Using python and gtk installed from homebrew :

```
$ brew install zoidbergwill/python/python35
$ brew install pygobject3 --with-python3 gtk+3 adwaita-icon-theme
```

Using the develop branch + https://github.com/pyinstaller/pyinstaller/pull/2596 changes from @virtuald 

```
$ git clone https://github.com/pyinstaller/pyinstaller
$ cd pyinstaller
$ git log -1 --format="%h"
abfc8066
$ git remote add virtuald https://github.com/virtuald/pyinstaller
$ git remote update
$ git merge virtuald/gtk-osx-fix
$ pip3 install .
```

With this sample script [test_img.tar.gz](https://github.com/pyinstaller/pyinstaller/files/1104967/test_img.tar.gz) :
```
$ test_img.py test_img.png
[works]
$ pyinstaller --log-level=TRACE -D -w test_img.py
... (see trace.txt)
$ dist/test_img/test_img test_img.png

(test_img:4209): Gtk-WARNING **: Locale not supported by C library.
	Using the fallback 'C' locale.

(test_img:4209): Gtk-WARNING **: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:gtkiconhelper.c:493:ensure_surface_for_gicon: assertion failed: (destination)
Abort trap: 6
```
[trace.txt](https://github.com/pyinstaller/pyinstaller/files/1104966/trace.txt)

From what I can see, pixbuf loaders are correctly embedded (`dist/test_img/lib/gdk-pixbuf-2.0/2.10.0/loaders`) but gdk-pixbuf fails somehow to load them.
But maybe the problem is elsewhere ...

**Comment by htgoebel** [comment]
You missed to post your *minimal* example *here*. We are not going to download and unpack an archive just to have a glance at your script. Also this does not leave the impression of being a *minimal* example.

Did you follow https://github.com/pyinstaller/pyinstaller/wiki/Recipe-Gtk-Application

**Comment by mgautierfr** [comment]
> You missed to post your minimal example here. We are not going to download and unpack an archive just to have a glance at your script. Also this does not leave the impression of being a minimal example.

```
#!/usr/bin/env python3

import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk

import sys

img_path = sys.argv[1]

print('try to display', img_path)

window = Gtk.Window()
window.connect("delete-event", Gtk.main_quit)

image = Gtk.Image()
image.set_from_file(img_path)

window.add(image)
window.show_all()
Gtk.main()
```

Here it is. I can difficultly make less to display an image using Gtk.

> Did you follow https://github.com/pyinstaller/pyinstaller/wiki/Recipe-Gtk-Application

I've miss this documentation. 
However, I've check what was include in the packaged application and loaders, pixbuf and pango rthooks are presents. (And hooks are runn at application startup)

This seems confirmed by the first note of the document :
```
Note: The current version of pyinstaller should have the appropriate runtime hooks builtin, so that these steps are unnecessary when building a GTK3 app that uses PyGObject.
```

**Comment by htgoebel** [comment]
Thanks for the example. The error message talks about an icon. Have you tried just opening an empty window (not trying to display an image)? Just to make sure searching in the correct direction. I don't have much time now, but may another ticket or pull-request might give more insights.

**Comment by mgautierfr** [comment]
> Have you tried just opening an empty window (not trying to display an image)?

With this script (image stuff commented):
```
#!/usr/bin/env python3

import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk

import sys

#img_path = sys.argv[1]

#print('try to display', img_path)

window = Gtk.Window()
window.connect("delete-event", Gtk.main_quit)

#image = Gtk.Image()
#image.set_from_file(img_path)

#window.add(image)
window.show_all()
Gtk.main()
```

I've no error (just the warning about the 'C' local).
An empty window is displayed and every thing seems to work.

> I don't have much time now, but may another ticket or pull-request might give more insights.

I would have love to give you a PR but after 1 full day of investigation I didn't find what is the problem.
I'm not a macOS expert so I give up and transfer the bug to some who may better fix this :)

**Comment by mgautierfr** [comment]
Hi @htgoebel, had you time to have a look on this issue ?

It seems pretty important as it is simply impossible to display a image using GTK on OSX. 

We (https://github.com/kiwix project) have few money we can spend on this bug and it is important for us.
If you (@htgoebel) or @virtuald (who already worked on GTK and OSX) (or others) can spend few time on this issue we can remunerate you for the fix.

**Comment by virtuald** [comment]
I like compensation. I can probably look at it later today.

**Comment by virtuald** [comment]
This should be fixed in the updated #2596 PR; try it out and let me know how it works.
